### PR TITLE
Add blobs as oracle records

### LINKED ISSUE #2032: Support blob records in blocks (see oracle records)

### PR DESCRIPTION
**Pull request by andresilva91**
## Motivation

We'll need blobs are oracles to be able to properly check things for #2151. Fixes #2032

## Proposal

Add blobs as oracles

## Test Plan

CI + will be used in following PRs

### CODE REVIEW

**Comment by andresilva91** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/linera-io/linera-protocol/2162?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#2162** <a href="https://app.graphite.dev/github/pr/linera-io/linera-protocol/2162?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#2158** <a href="https://app.graphite.dev/github/pr/linera-io/linera-protocol/2158?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#2156** <a href="https://app.graphite.dev/github/pr/linera-io/linera-protocol/2156?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @andresilva91 and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: linera-base/src/data_types.rs [hunk 1]
```diff
@@ -725,6 +725,8 @@ pub enum OracleResponse {
     Service(Vec<u8>),
     /// The response from an HTTP POST request.
     Post(Vec<u8>),
+    /// The response from a blob interaction.
```
Comment (afck) [inline linera-base/src/data_types.rs:728]: Maybe:

```suggestion
    /// A successful read or write of a blob.
```

**Review by afck — APPROVED** [review]

File: linera-base/src/data_types.rs [hunk 2]
```diff
@@ -718,24 +718,45 @@ pub struct OracleRecord {
     pub responses: Vec<OracleResponse>,
 }
 
+impl OracleRecord {
+    /// Wether an `OracleRecord` is permitted in fast blocks or not.
+    pub fn is_permitted_in_fast_blocks(&self) -> bool {
+        self.responses.is_empty()
+            || self
```
Comment (afck) [inline linera-base/src/data_types.rs:725]: `all` is `true` for an empty iterator.

```suggestion
        self
```

**Review by afck — APPROVED** [review]

File: linera-base/src/data_types.rs (hunk 2 above)
Comment (andresilva91) [reply linera-base/src/data_types.rs:725]: Ah nice ok, good to know

**Pull request merged**